src/src/camp/camp.service.ts

```diff
@@ -3646,87 +3646,69 @@ export class CampService {
 
 	public async getRandomAttendanceGeneration(id: any, req: any, res: any) {
 		let camp_id = id;
-		const currentDate = new Date().toISOString().split('T')[0];
-		const currentDateObject = new Date(currentDate);
+		const format = 'YYYY-MM-DD';
+		const currentDate = moment().format(format);
 
 		// Get the date 7 days before the current date
-		const sevenDaysLess = new Date(currentDateObject);
-		sevenDaysLess.setDate(currentDateObject.getDate() - 7);
-		const formattedSevenDaysLess = sevenDaysLess
-			.toISOString()
-			.split('T')[0]; // Format seven days less
-
-		// Get the date 1 day before the current date
-		const oneDayLess = new Date(currentDateObject);
-		oneDayLess.setDate(currentDateObject.getDate() - 1);
-		const formattedOneDayLess = oneDayLess.toISOString().split('T')[0];
-
-		const sql = `SELECT
-		gs.date AS date,
-		COUNT(a.id) AS count_of_data
-	  FROM
-		generate_series(
+		const formattedSevenDaysLess = moment()
+			.subtract(7, 'days')
+			.format(format); // Format seven days less
+
+		const sql = `SELECT gs.date AS date, COUNT(a.id) AS count_of_data
+	  FROM generate_series(
 		  '${formattedSevenDaysLess}'::timestamp,
-		  '${formattedOneDayLess}'::timestamp,
+		  '${currentDate}'::timestamp,
 		  '1 day'
 		) AS gs(date)
-	  LEFT JOIN
-		attendance a ON DATE_TRUNC('day', a.date_time) = gs.date AND a.photo_1 IS NOT NULL AND a.context_id = ${camp_id}
-	  WHERE
-		gs.date >= '${formattedSevenDaysLess}' AND gs.date <= '${currentDate}'
-	  GROUP BY
-		gs.date
-	  ORDER BY
-		gs.date;
-	  
+	  LEFT JOIN	attendance a ON DATE_TRUNC('day', a.date_time) = gs.date 
+		LEFT JOIN camp_days_activities_tracker cda ON cda.id = a.context_id
+	  WHERE gs.date >= '${formattedSevenDaysLess}' AND gs.date <= '${currentDate}'
+		AND a.photo_1 IS NOT NULL AND a.photo_1 != '-'
+		AND cda.camp_id = ${camp_id}
+		AND a.context = 'camp_days_activities_tracker'
+	  GROUP BY gs.date
+	  ORDER BY gs.date;
 		;`;
+
 		const attendance_data = (
 			await this.hasuraServiceFromServices.executeRawSql(sql)
 		)?.result;
```

Comment:
Add try catch and in catch return query error.

Reply:
resolved